packages/@react-spectrum/s2/test/SelectBoxGroup.test.tsx

```diff
@@ -66,44 +84,187 @@ function DisabledSelectBox() {
 }
 
 describe('SelectBoxGroup', () => {
+  let user;
+
+  beforeAll(function () {
+    jest.useFakeTimers();
+    jest.spyOn(window.HTMLElement.prototype, 'clientWidth', 'get').mockImplementation(() => 100);
+    jest.spyOn(window.HTMLElement.prototype, 'clientHeight', 'get').mockImplementation(() => 100);
+    jest.spyOn(window.HTMLElement.prototype, 'scrollHeight', 'get').mockImplementation(() => 50);
+    user = userEvent.setup({delay: null, pointerMap});
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    act(() => jest.runAllTimers());
+  });
+
+  afterAll(function () {
+    jest.restoreAllMocks();
+  });
+
   describe('Basic functionality', () => {
     it('renders as a listbox with options', () => {
-      render(<SingleSelectBox />);
+      render(<ControlledSingleSelectBox />);
       expect(screen.getByRole('listbox')).toBeInTheDocument();
       expect(screen.getAllByRole('option')).toHaveLength(3);
       expect(screen.getByText('Option 1')).toBeInTheDocument();
     });
 
     it('renders multiple selection mode', () => {
-      render(<MultiSelectBox />);
+      render(<ControlledMultiSelectBox />);
       expect(screen.getByRole('listbox')).toBeInTheDocument();
       expect(screen.getAllByRole('option')).toHaveLength(3);
       expect(screen.getByText('Option 1')).toBeInTheDocument();
     });
+  });
 
-    it('handles selection in single mode', async () => {
-      render(<SingleSelectBox />);
+  describe('Uncontrolled behavior', () => {
+    it('handles uncontrolled click selection in single mode', async () => {
+      render(<UncontrolledSelectBox selectionMode="single" />);
       const options = screen.getAllByRole('option');
       const option1 = options.find(option => option.textContent?.includes('Option 1'))!;
       
-      await userEvent.click(option1);
+      await user.click(option1);
```

Comment:
We can probably make use of the test-util `ListBoxTester` in order to do interactions and avoid things like
```
      const options = screen.getAllByRole('option');
      const option1 = options.find(option => option.textContent?.includes('Option 1'))!;
```
https://react-spectrum.adobe.com/react-aria/ListBox.html#testing

These three lines would be something like
```
toggleOptionSelection({option: 1})
```

Then it's easier to write tests which are agnostic of keyboard or mouse, so you can run both on the same series of tests